docker/django/Dockerfile

```diff
@@ -32,28 +32,19 @@ ENV POETRY_VERSION=1.6.1 \
     # do not ask any interactive question
     POETRY_NO_INTERACTION=1
 
-# install poetry - respects $POETRY_VERSION & $POETRY_HOME
-RUN curl -sSL https://install.python-poetry.org | python3 -
-
-ARG POETRY_HOME
-
 ENV PYTHONUNBUFFERED=1 \
-    # paths
-    # where to create the env
-    VENV_PATH="/opt/pysetup/.venv" \
-    # this is where our requirements + virtual environment will live
+    # this is where our requirements are copied to
     PYSETUP_PATH="/opt/pysetup"
 
-# prepend poetry and venv to path
-ENV PATH="/opt/poetry/bin:$VENV_PATH/bin:$PATH"
```

Comment:
`/opt/poetry/bin` is unnecessary if we directly address `$POETRY_HOME/bin/poetry`. On balance, the `$PATH` convenience isn’t compelling.

…One of the key differences between the script install and the `pip` install is that Poetry’s `venv` is one step removed from `$POETRY_HOME`, at `/opt/poetry/venv`. And `/opt/poetry/bin` is a directory with only one file, `poetry` that is a symlink to `/opt/poetry/venv/bin/poetry`. After this change, Poetry’s `venv` _is_ `/opt/poetry` and this less conducive to `$PATH` (kicking it to the end would be the fix `PATH="$PATH:$POETRY_HOME/bin"`).

A comparison of the `$POETRY_HOME` structure:

<table width="100%">
<tr>
<th>

script

</th>
<th>

`pip`

</th>
</tr>
<tr>
<th colspan="2">

```
% docker run -it freelawproject/courtlistener:main-web-dev find /opt/poetry -maxdepth 2
```

</th>
</tr>
<tr>
<td valign="top" width="50%">

```
/opt/poetry
/opt/poetry/VERSION
/opt/poetry/venv
/opt/poetry/venv/pyvenv.cfg
/opt/poetry/venv/include
/opt/poetry/venv/lib
/opt/poetry/venv/lib64
/opt/poetry/venv/poetry_env
/opt/poetry/venv/bin
/opt/poetry/bin
/opt/poetry/bin/poetry
```

</td>
<td valign="top" width="50%">

```
/opt/poetry
/opt/poetry/pyvenv.cfg
/opt/poetry/include
/opt/poetry/include/python3.11
/opt/poetry/lib
/opt/poetry/lib/python3.11
/opt/poetry/lib64
/opt/poetry/bin
/opt/poetry/bin/doesitcache
/opt/poetry/bin/python
/opt/poetry/bin/activate.csh
/opt/poetry/bin/jsonschema
/opt/poetry/bin/virtualenv
/opt/poetry/bin/pkginfo
/opt/poetry/bin/dulwich
/opt/poetry/bin/python3.11
/opt/poetry/bin/pip3
/opt/poetry/bin/activate
/opt/poetry/bin/dul-upload-pack
/opt/poetry/bin/pip3.11
/opt/poetry/bin/Activate.ps1
/opt/poetry/bin/normalizer
/opt/poetry/bin/dul-receive-pack
/opt/poetry/bin/pyproject-build
/opt/poetry/bin/python3
/opt/poetry/bin/pip
/opt/poetry/bin/activate.fish
/opt/poetry/bin/poetry
/opt/poetry/bin/keyring
```

</td>
</tr>
</table>